pkgs/by-name/va/vatprism/package.nix

```
@@ -0,0 +1,109 @@
+# Made in accordance to https://fzakaria.com/2020/07/20/packaging-a-maven-application-with-nix.html
+{ lib
+, stdenv
+, jdk
+, maven
+, makeWrapper
+, fetchFromGitHub
+, libGL
+, libxkbcommon
+, wayland
+, fontconfig
+, xorg
+}: let
+  pname = "vatprism";
+  version = "0.3.5";
+  src = fetchFromGitHub {
+    owner = "marvk";
+    repo = "vatprism";
+    rev = "v${version}";
+    hash = "sha256-ofEwHUCm79roHe2bawmKFw2QHhIonnlkFG5nhE6uN+g=";
+  };
+  dependencies = stdenv.mkDerivation {
+    name = "${pname}-dependencies1";
+    nativeBuildInputs = [ jdk maven ];
+    # https://github.com/marvk/vatprism/pull/141
+    patches = [ ./0001-Fix-build-on-JDK-21.patch ];
+    inherit src;
+    buildPhase = ''
+      runHook preBuild
+      while mvn package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+        echo "timeout, restart maven to continue downloading"
+      done
+      runHook postBuild
+    '';
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      runHook preInstall
+      find $out/.m2 -type f -regex '.+\\(\\.lastUpdated\\|resolver-status\\.properties\\|_remote\\.repositories\\)' -delete
+      runHook postInstall
+    '';
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "zHPSFz8GggFWxrIkJF2DmB1OI/y3UX1AOFBCaVvDK3Y=";
+  };
+  libPath = lib.makeLibraryPath [
+    libGL
+    libxkbcommon
+    wayland
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXxf86vm
+    xorg.libXtst
+    fontconfig
+  ];
+in
+  stdenv.mkDerivation rec {
+    inherit pname version src;
+
+    buildInputs = [ jdk ];
+    nativeBuildInputs = [ maven makeWrapper ];
+    # https://github.com/marvk/vatprism/pull/141
+    patches = [ ./0001-Fix-build-on-JDK-21.patch ];
+
+    buildPhase = ''
+      runHook preBuild
+      # 'maven.repo.local' must be writable so copy it out of nix store
+      mvn package --offline -Dmaven.repo.local=${dependencies}/.m2
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      # create the bin directory
+      mkdir -p $out/bin
+
+      # create a symbolic link for the lib directory
+      ln -s ${dependencies}/.m2 $out/lib
+
+      # copy out the JAR
+      # Maven already setup the classpath to use m2 repository layout
+      # with the prefix of lib/
+      cp target-fat-jar/vatsim-map-${version}-fat.jar $out/
+
+      # create a wrapper that will automatically set the classpath
+      # this should be the paths from the dependency derivation
+      makeWrapper ${jdk}/bin/java $out/bin/${pname} \
+          --add-flags "-jar $out/vatsim-map-${version}-fat.jar" \
+          --set JAVA_HOME ${jdk.home} \
+          --suffix LD_LIBRARY_PATH : ${libPath}
+      runHook postInstall
+    '';
+
+    meta = with lib; {
```

```suggestion
    meta = {
```

tracking issue: #292468